doc/specs/stdlib_math.md

```diff
@@ -91,6 +91,52 @@ program demo_clip_real
 end program demo_clip_real
 ```
 
+### `gcd` function
+
+#### Description
+
+Returns the greatest common divisor of two integers.
+
+#### Syntax
+
+`res = [[stdlib_math(module):gcd(interface)]] (a, b)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`a`: An integer.
+`b`: Another integer.
+
+Note: All arguments must be integers of the same `kind`.
+
+#### Output value or Result value
+
+The output is a scalar integer of the same `kind` as to that of the arguments.
+
+#### Examples
+
+##### Example 1:
+
+```fortran
+program demo_gcd
+  use stdlib_math, only: gcd
+  implicit none
+  integer :: a, b, c
```

Comment:
Should `kind` be mentioned explicitly here to emphasize the point made in line 117?

Reply:
There was explicit kinds here initially but I was asked to remove them since int32 is the default kind.